src/math/MultiplierPointMath.sol

```diff
@@ -0,0 +1,166 @@
+// SPDX-License-Identifier: MIT-1.0
+pragma solidity ^0.8.26;
+
+import { Math } from "@openzeppelin/contracts/utils/math/Math.sol";
+import { IStakeConstants } from "../interfaces/IStakeConstants.sol";
+
+/**
+ * @title MultiplierPointMath
+ * @author Ricardo Guilherme Schmidt <ricardo3@status.im>
+ * @notice Provides mathematical operations and utilities for managing multiplier points in the staking system.
+ */
+abstract contract MultiplierPointMath is IStakeConstants {
+    /// @notice One (mean) tropical year, in seconds.
+    uint256 public constant YEAR = 365 days;
+    /// @notice Accrued multiplier points maximum multiplier.
+    uint256 public constant MAX_MULTIPLIER = 4;
+    /// @notice Multiplier points annual percentage yield.
+    uint256 public constant MP_APY = 100;
+    /// @notice Multiplier points accrued maximum percentage yield.
+    uint256 public constant MP_MPY = MAX_MULTIPLIER * MP_APY;
+    /// @notice Multiplier points absolute maximum percentage yield.
+    uint256 public constant MP_MPY_ABSOLUTE = 100 + (2 * (MAX_MULTIPLIER * MP_APY));
+    /// @notice The accrue rate period of time over which multiplier points are calculated.
+    uint256 public constant ACCRUE_RATE = 12 seconds;
```

Comment:
How do we guarantee that this rate is always 12 seconds?

Reply:
It's not guaranteed, it's the minimal possible accrue_rate possible. If one to call the contract every block, or at least 2 consecutive blocks, as the blocktime is 12 seconds, this would be the accure rate.

Reply:
> or at least 2 consecutive blocks, as the blocktime is 12 seconds

We get faster soft confirmations on L2s tho. 
The same has been discussed here https://github.com/vacp2p/staking-reward-streamer/issues/52#issuecomment-2554656433

Not sure if you ever followed up on the changes but back then, we fixed it by ensuring we only increase the `lastUpdatedTime` only if MPs were actually generated.

This way, you don't need to know about any block time at all.

Here are the changes we've made back then: https://github.com/vacp2p/staking-reward-streamer/pull/100/files#diff-b8371c7492fb939a38fb61c84fb63d70e61f70c633e1a09e6e059c8372ec3846R493-R496

What do you think?